### PR TITLE
Allow mirrord to be enabled or disabled by default

### LINKED ISSUE #268: Allow to configure the default enabled state of mirrord
Same as [here](https://github.com/metalbear-co/mirrord-vscode/issues/131)

### PR DESCRIPTION
**Pull request by gememma**
Closes #268

### CODE REVIEW

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt [hunk 1]
```diff
@@ -23,7 +29,20 @@ class MirrordEnabler : ToggleAction(), DumbAware {
...
+        val state = e.project?.service<MirrordProjectService>()?.enabled ?: false
+        e.presentation.icon = if (state) MirrordIcons.enabled else MirrordIcons.disabled
 
         super.update(e)
     }
+
+    override fun runActivity(project: Project) {
+        if (instance.mirrordState.enabledByDefault) {
+            project.service<MirrordProjectService>().enabled = true
+            ActivityTracker.getInstance().inc()
```
Comment (Razz4780) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt:40]: What's this? :eyes:

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt [hunk 2]
```diff
@@ -23,7 +29,20 @@ class MirrordEnabler : ToggleAction(), DumbAware {
...
+            project.service<MirrordProjectService>().notifier.notification(
+                "mirrord is enabled by default",
+                NotificationType.IDE_UPDATE
+            )
```
Comment (Razz4780) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt:44]: No need to notify, I think

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordDropDown.kt [hunk 3]
```diff
@@ -33,6 +34,30 @@ class MirrordDropDown : ComboBoxAction(), DumbAware {
...
+                    "mirrord set to be disabled by default",
+                    NotificationType.IDE_UPDATE
+                )
+            }
+
+            notification.fire()
+
+            return
+        }
+    }
```
Comment (Razz4780) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordDropDown.kt:59]: No need for this, I think. The user can change this in the settings

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt (hunk 1 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt:40]: Huh, nice. Thanks!

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordDropDown.kt (hunk 3 above)
Comment (gememma) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordDropDown.kt:59]: no item in the dropdown menu at all?

Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordDropDown.kt:59]: Yup

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordSettingsComponent.kt [hunk 4]
```diff
@@ -17,6 +17,7 @@ class MirrordSettingsComponent {
         .associateWith { JBCheckBox(it.presentableName) }
 
     private val usageBannerEnabled = JBCheckBox("Show usage banner on startup")
+    private val enabledOnStartup = JBCheckBox("Enable mirrord by default")
```
Comment (Razz4780) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordSettingsComponent.kt:20]:
```suggestion
    private val enabledOnStartup = JBCheckBox("Enable mirrord on startup")
```

**Review by Razz4780 — APPROVED** [review]
Neat! One suggestion only

**Pull request merged**